MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/DataProcessorGenerateNotebook.h

```diff
@@ -0,0 +1,150 @@
+#ifndef MANTID_CUSTOMINTERFACES_DATAPROCESSORGENERATENOTEBOOK_H
+#define MANTID_CUSTOMINTERFACES_DATAPROCESSORGENERATENOTEBOOK_H
+
+/** @class DataProcessorGenerateNotebook
+
+    This class creates ipython notebooks from the ISIS Reflectometry
+    (Polref) interface
+
+    Copyright &copy; 2007-8 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+    National Laboratory & European Spallation Source
+
+    This file is part of Mantid.
+
+    Mantid is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    Mantid is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+    File change history is stored at: <https://github.com/mantidproject/mantid>.
+    Code Documentation is available at: <http://doxygen.mantidproject.org>
+    */
+
+#include "MantidKernel/System.h"
+#include "MantidQtCustomInterfaces/Reflectometry/DataPostprocessorAlgorithm.h"
+#include "MantidQtCustomInterfaces/Reflectometry/DataPreprocessorAlgorithm.h"
+#include "MantidQtCustomInterfaces/Reflectometry/DataProcessorAlgorithm.h"
+#include "MantidQtCustomInterfaces/Reflectometry/DataProcessorWhiteList.h"
+#include "MantidQtCustomInterfaces/Reflectometry/QDataProcessorTableModel.h"
+
+#include <boost/tuple/tuple.hpp>
+#include <map>
+#include <set>
+#include <sstream>
+#include <string>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+std::vector<std::string>
+    DLLExport splitByCommas(const std::string &names_string);
+
+std::string DLLExport plot1DString(const std::vector<std::string> &ws_names);
+
+std::string DLLExport tableString(QDataProcessorTableModel_sptr model,
+                                  const DataProcessorWhiteList &whitelist,
+                                  const std::set<int> &rows);
+
+std::string DLLExport titleString(const std::string &wsName);
+
+boost::tuple<std::string, std::string> DLLExport postprocessGroupString(
+    const std::set<int> &rows,
```

Comment:
Too many arguments to function. Can you group these perhaps putting related arguments into a single `struct`.